tests/components/opnsense/test_device_tracker.py

```diff
@@ -0,0 +1,35 @@
+"""The tests for the opnsense device tracker platform."""
+
+from unittest import mock
+
+from homeassistant.components.opnsense import CONF_API_SECRET, DOMAIN, OPNSENSE_DATA
+from homeassistant.const import CONF_URL, CONF_API_KEY, CONF_VERIFY_SSL
+from homeassistant.setup import async_setup_component
+
+from tests.common import MockDependency
+
+
+async def test_get_scanner(hass):
+    """Test creating an opnsense scanner."""
+    with MockDependency("pyopnsense") as mocked_opnsense:
+        get_arp = mock.MagicMock()
```

Comment:
If we set the `return_value` for this mock to some mock device data with mac address etc, we can test that the state is updated correctly.

Reply:
Done in: https://github.com/home-assistant/home-assistant/pull/26834/commits/cd1bbe60d0140a59fdd547fc35db8b1747afb36a